db/blob/blob_source_test.cc

```diff
@@ -1020,6 +1022,299 @@ TEST_F(BlobSourceTest, MultiGetBlobsFromCache) {
   }
 }
 
+class BlobSecondaryCacheTest : public DBTestBase {
+ protected:
+ public:
+  explicit BlobSecondaryCacheTest()
+      : DBTestBase("blob_secondary_cache_test", /*env_do_fsync=*/true) {
+    options_.env = env_;
+    options_.enable_blob_files = true;
+    options_.create_if_missing = true;
+
+    // Set a small cache capacity to evict entries from the cache, and to test
+    // that secondary cache is used properly.
+    lru_cache_ops_.capacity = 1024;
+    lru_cache_ops_.num_shard_bits = 2;
+    lru_cache_ops_.metadata_charge_policy = kDontChargeCacheMetadata;
+
+    secondary_cache_opts_.capacity = 8 << 20;  // 8 MB
+    secondary_cache_opts_.num_shard_bits = 2;
+    secondary_cache_opts_.metadata_charge_policy = kDontChargeCacheMetadata;
+
+    // Read blobs from the secondary cache if they are not in the primary cache
+    options_.lowest_used_cache_tier = CacheTier::kNonVolatileBlockTier;
+
+    assert(db_->GetDbIdentity(db_id_).ok());
+    assert(db_->GetDbSessionId(db_session_id_).ok());
+  }
+
+  Options options_;
+
+  LRUCacheOptions lru_cache_ops_;
+  CompressedSecondaryCacheOptions secondary_cache_opts_;
+
+  std::string db_id_;
+  std::string db_session_id_;
+};
+
+TEST_F(BlobSecondaryCacheTest, GetBlobsFromSecondaryCache) {
+  if (!Snappy_Supported()) {
+    return;
+  }
+
+  secondary_cache_opts_.compression_type = kSnappyCompression;
+  lru_cache_ops_.secondary_cache =
+      NewCompressedSecondaryCache(secondary_cache_opts_);
+  options_.blob_cache = NewLRUCache(lru_cache_ops_);
+
+  options_.cf_paths.emplace_back(
+      test::PerThreadDBPath(
+          env_, "BlobSecondaryCacheTest_GetBlobsFromSecondaryCache"),
+      0);
+
+  options_.statistics = CreateDBStatistics();
+  Statistics* statistics = options_.statistics.get();
+  assert(statistics);
+
+  DestroyAndReopen(options_);
+
+  ImmutableOptions immutable_options(options_);
+
+  constexpr uint32_t column_family_id = 1;
+  constexpr bool has_ttl = false;
+  constexpr ExpirationRange expiration_range;
+  constexpr uint64_t file_number = 1;
+
+  Random rnd(301);
+
+  std::vector<std::string> key_strs;
+  std::vector<std::string> blob_strs;
+  key_strs.push_back("key0");
+  blob_strs.push_back(rnd.RandomString(1020));
+  key_strs.push_back("key1");
+  blob_strs.push_back(rnd.RandomString(1020));
+
+  std::vector<Slice> keys;
+  std::vector<Slice> blobs;
+  keys.push_back({key_strs[0]});
+  blobs.push_back({blob_strs[0]});
+  keys.push_back({key_strs[1]});
+  blobs.push_back({blob_strs[1]});
```

Comment:
Minor but these vectors could be initialized using the braced initialization syntax (without having to do `push_back`s)